aptos-move/block-executor/src/view.rs

```diff
@@ -366,7 +452,8 @@ impl<'a, T: Transaction, S: TStateView<Key = T::Key>, X: Executable> TResourceVi
         &self,
         state_key: &Self::Key,
     ) -> anyhow::Result<Option<StateValueMetadataKind>> {
-        self.get_resource_state_value_impl(state_key, ReadKind::Metadata)
+        // TODO check that passing None here is correct
+        self.get_resource_state_value_impl(state_key, None, ReadKind::Metadata)
```

Comment:
@gelash also these changes - we don't need layout for non-value read kinds, right? (as we don't do exchanges)

Reply:
we shouldn't